www/src/pages/editor/NewMineralSiteModal.tsx

```diff
@@ -0,0 +1,442 @@
+import { useStores, Commodity, DraftCreateMineralSite, CandidateEntity, DepositType, Unit, MineralSite } from "models";
+import { LocationInfo } from "../../models/mineralSite/LocationInfo";
+import { Reference, Document } from "../../models/mineralSite/Reference";
+import { GradeTonnage } from "../../models/mineralSite/GradeTonnage";
+import { CommodityStore } from "models/commodity";
+import axios from "axios";
+import { Country, StateOrProvince } from "models";
+import { useEffect, useState } from "react";
+import { v4 as uuidv4 } from "uuid";
+import { UserStore } from "models/user";
+import { MineralInventory, Measure } from "../../models/mineralSite/MineralInventory";
+import { useMemo } from "react";
+
+interface NewMineralSiteModalProps {
+  commodity: Commodity;
+  visible: boolean;
+  onClose: () => void;
+}
+interface FormValues {
+  latitude?: number;
+  longitude?: number;
+  country?: string;
+  stateOrProvince?: string;
+  name: string;
+  refDoc: string;
+  refComment?: string;
+  sourceType: string;
+  depositType: string;
+  depositTypeConfidence: number;
+  tonnage?: number;
+  grade?: number;
+  gradeUnit?: string;
+  tonnageUnit?: string;
+}
+
+export const NewMineralSiteModal: React.FC<NewMineralSiteModalProps> = ({ commodity, visible, onClose }) => {
+  const { mineralSiteStore, dedupMineralSiteStore, userStore, commodityStore, countryStore, stateOrProvinceStore, depositTypeStore, unitStore } = useStores();
+  const [form] = Form.useForm();
+  const [recordId, setRecordId] = useState("");
+  const [selectedSourceType, setSelectedSourceType] = useState<string | null>(null);
+  const commodityOptions = useMemo(() => {
+    return Array.from(commodityStore.records.values()).map((comm) => ({
+      value: comm?.id,
+      label: comm?.name,
+    }));
+  }, [commodityStore.records]);
+  const currentUser = userStore.getCurrentUser()?.url;
+  const countryOptions = useMemo(() => {
+    return Array.from(countryStore.records.values()).map((country) => ({
+      value: country?.id,
+      label: country?.name,
+    }));
+  }, [countryStore.records]);
+
+  const stateOptions = useMemo(() => {
+    return Array.from(stateOrProvinceStore.records.values()).map((state) => ({
+      value: state?.id,
+      label: state?.name,
+    }));
+  }, [stateOrProvinceStore.records]);
+
+  const depositTypeOptions = useMemo(() => {
+    return Array.from(depositTypeStore.records.values()).map((type) => ({
+      value: type?.id ?? "",
+      label: type?.name ?? "Unnamed",
+    }));
+  }, [depositTypeStore.records]);
+
+  const unitOptions = useMemo(() => {
+    return Array.from(unitStore.records.values()).map((unit) => ({
+      value: unit?.id ?? "",
+      label: unit?.name ?? "Unnamed",
+    }));
+  }, [unitStore.records]);
+  useEffect(() => {
+    const initialize = async () => {
+      try {
+
+        if (visible) {
+          const generatedId = `record-${uuidv4()}`;
```

Comment:
again, I think there is an issue with this approach. you should create a record Id at the time of submission instead of useEffect for this.